anthos-bm-gcp-terraform/docs/nfs.md

```diff
@@ -1,87 +1,64 @@
-# download the NFS CSI driver installation script and run it
-# the script configures your cluster to support the NFS storage
-curl -skSL https://raw.githubusercontent.com/kubernetes-csi/csi-driver-nfs/v3.1.0/deploy/install-driver.sh | bash -s v3.1.0 --
-# create a new StorageClass for NFS storage
-kubectl apply -f $HOME/nfs-csi.yaml
-```
-
-3. Verify the available storage classes
-```sh
-kubectl get storageclass
-
-# expected output should include
-NAME            PROVISIONER                    RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
-nfs-csi         nfs.csi.k8s.io                 Delete          Immediate              false                  5m
-```
+### Steps
 
-4. The `nfs-csi` storage class can now be utilized for your `PersistantVolumeClaims`.
-```sh
-# example for a DataVolume
-...
-spec:
-  pvc:
-    storageClassName: nfs-csi
-...
-```
+1. Follow any of the guides *([Quick starter](quickstart.md), [All in one install](one_click_install.md) or [Manual LB install](manuallb_install.md))* with one additional variable added to the `terraform.tfvars` file.
 
-### All in one or Manual LB install
+    ```sh
+    # terraform.tfvars file
+    ...
+    nfs_server = true
+    ...
+    ```
+2. The [All in one install](one_click_install.md) and [Manual LB install](manuallb_install.md) guides will automatically setup everything required for the NFS storage class to be available.
 
-1. Follow the [All in one install](one_click_install.md) or [Manual LB install](manuallb_install.md) guides except additionally add the following variable to your `terraform.tfvars` file.
-```sh
-# terraform.tfvars file
-...
-nfs_server = true
-...
-```
+    However, for the **[Quick starter](quickstart.md)** mode installation, you will see some additional steps _(in addition to the common ones from the     Quick starter)_, printed out as part of the *Terraform output*. Run those commands from inside your admin-workstation to install the `NFS Driver`       into your cluster.
 
-With `nfs_server` set to `true`, the *All in one* and *Manual LB install* modes automatically configure everything required for the `NFS CSI`.
+    SSH into your **admin workstation** and ensure that the `KUBECONFIG` environment variable is set to the path of your Anthos on bare metal cluster       context file.
+    ```sh
+    # example output for the Quick starter installation with 'nfs_server = true'
 
-2. Once the cluster is fully deployed, verify the available storage classes
-```sh
-kubectl get storageclass
+    ################################################################################
+    #     Configure the cluster to utilize NFS for PVs with the NFS-CSI driver     #
+    ################################################################################
+    > export KUBECONFIG=bmctl-workspace/cluster1/cluster1-kubeconfig && \
+      curl -skSL https://raw.githubusercontent.com/kubernetes-csi/csi-driver-nfs/v3.1.0/deploy/install-driver.sh | bash -s v3.1.0 -- && \
```

Comment:
just wondering if we had to update the version that we point to, or is it ok if we point to main?

Reply:
If you are comfortable point to HEAD, here is the command:

`curl -skSL https://raw.githubusercontent.com/kubernetes-csi/csi-driver-nfs/master/deploy/install-driver.sh | bash -s master --`

This should be updated to match: https://github.com/GoogleCloudPlatform/anthos-samples/blob/51af41a243f6b7423d5568db774580944004fe24/anthos-bm-gcp-terraform/resources/install_abm.sh#L49

Reply:
I think it's better to pin the version to a tested and working one than pointing to the HEAD. Probably we should update the versions regularly as part of the maintenance of this sample. It could be done during the release process too.

Reply:
ok cool. We should probably outline for us how often we want to update the versions! Ok as. long as its noted somewhere, LGTM! Thanks both!

Reply:
Good point. Documented it in a followup PR: https://github.com/GoogleCloudPlatform/anthos-samples/pull/319